Automattic/jetpack
# Admin Menu: Second pass at menu structure classes

Let's see if we can break out the special cases into their own subclasses. It would make each class easier to read and follow, and it would be easier to make adjustments in an environment-based way.

#### Changes proposed in this Pull Request:

* Splits out WP.com and Atomic-specific code from `Admin_Menu` class
* Adjusts tests to cover the respective environments

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






It's not sure easy to test because the point of the PR is that nothing changes :)

Jetpack:
* On your Jetpack site, enable the admin menu with `add_filter( 'jetpack_load_admin_menu_class', '__return_true' );`
* Make sure there is not a "Hosting Configuration" link under Settings
* Make sure there are no links to Earn and Marketing under Tools.

Atomic:
* On an Atomic site, enable Jetpack beta and switch to this branch
* Enable the admin menu with `add_filter( 'jetpack_load_admin_menu_class', '__return_true' );`
* Make sure there's still a "Hosting Configuration" link under Settings
* Make sure the Plugins menu item points to wp-admin/plugins.php

WP.com:
* Apply D54406-code and D54549-code to your sandbox.
* Make sure the Plugins menu item points to wordpress.com/plugins/site

#### Proposed changelog entry for your changes:


* None needed.
